cloudify/tests/test_decorators.py

```diff
@@ -184,3 +188,110 @@ def w2():
         self.assertEqual(o2(), 'o2')
         self.assertEqual(w1(), 'w1')
         self.assertEqual(w2(), 'w2')
+
+    @patch('cloudify.tests.mocks.mock_rest_client.'
+           'MockNodeInstancesClient.list')
+    @patch('cloudify.decorators.get_rest_client')
+    def test_serial_operation_install(self, _mock, list_fn):
+
+        _ctx = MockCloudifyContext(
+            node_id='test_node',
+            deployment_id='test_deployment',
+            index=7
+        )
+
+        # Check that we raise if the preceder is not 'started'.
+        _ctx._context['workflow_id'] = 'install'
+        current_ctx.set(_ctx)
+
+        _mock.side_effect = lambda: rest_client_mock.MockRestclient()
+
+        @serial_operation()
+        def run_op_install_only(**kwargs):
+            run(assert_op_impl, **kwargs)
+
+        @serial_operation(revert_to_parallel_threshold=1)
+        def run_op_wait_for_1(**kwargs):
+            run(assert_op_impl, **kwargs)
+
+        @serial_operation(states=['uninitialized'])
+        def run_op_uninitialized_states(**kwargs):
+            run(assert_op_impl, **kwargs)
+
+        @serial_operation(workflows=['scale'])
+        def run_op_neither(**kwargs):
+            run(assert_op_impl, **kwargs)
+
+        list_fn.return_value = [
+            MagicMock(id='test_node{x}'.format(x=n),
```

Comment:
could well be just `Mock`, no reason this has to be `MagicMock`

Reply:
OK